update mwc-slider to MDC 4

e111077 opened the pull request:
fixes #486.

Also cleaned up innerText implementation and cleaned up render function to look less ugly.

azakus commented on packages/slider/src/mwc-slider-base.ts:
```diff
@@ -174,10 +170,21 @@ export class SliderBase extends FormElement {
+        const markerWidth = '2px';
+        const markerBkgdImage = `linear-gradient(to right, currentColor ${
+            markerWidth}, transparent 0)`;
+        const markerBkgdLayout = `0 center / calc((100% - ${markerWidth}) / ${
```
This is starting to get a little complicated. Can you put a comment in here about what this calculation is doing?

e111077 replied:
Sure, I hate it, but it's exactly what they do in the component. It basically draws the notches on the slider if you enable notches and steps. It's more performant, but ugly

e111077 replied:
done

e111077 commented:
Re-requesting review as I refactored it a bit to not use `element.style.setProperty` to use `styleMap`. Additionally, added `{passive: true}` to some things that were giving us warnings

azakus commented on packages/slider/src/mwc-slider-base.ts:
```diff
@@ -137,14 +146,18 @@ export class SliderBase extends FormElement {
-          <K extends EventType>(type: K, handler: SpecificEventListener<K>) =>
-              this.mdcRoot.addEventListener(type, handler),
+          <K extends EventType>(type: K, handler: SpecificEventListener<K>) => {
+            const init = type === 'touchstart' ? {passive: true} : {};
```
This should be using `applyPassive()` from `@material/dom/events` like we use in ripple-directive: https://github.com/material-components/material-components-web-components/blob/master/packages/ripple/src/ripple-directive.ts#L17

azakus commented on packages/slider/src/mwc-slider-base.ts:
```diff
@@ -137,14 +146,18 @@ export class SliderBase extends FormElement {
-          <K extends EventType>(type: K, handler: SpecificEventListener<K>) =>
-              this.thumbContainer.addEventListener(type, handler),
+          <K extends EventType>(type: K, handler: SpecificEventListener<K>) => {
+            const init = type === 'touchstart' ? {passive: true} : {};
```
`applyPassive()`

azakus commented on packages/slider/src/mwc-slider-base.ts:
```diff
@@ -155,7 +168,7 @@ export class SliderBase extends FormElement {
               document.body.removeEventListener(type, handler),
       registerResizeHandler: (handler: SpecificEventListener<'resize'>) =>
-          window.addEventListener('resize', handler),
+          window.addEventListener('resize', handler, {passive: true}),
```
`applyPassive()`

azakus commented on packages/slider/src/mwc-slider-base.ts:
```diff
@@ -127,6 +131,11 @@ export class SliderBase extends FormElement {
 
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    this.mdcFoundation.destroy();
```
Why destroy the foundation on `disconnectedCallback()`?

e111077 replied:
to remove all the body resize observers

azakus replied:
Seems like we either need an explicit removeEventListener wrapper in the foundation, or we'll need to recreate the foundation in `connectedCallback()`

e111077 replied:
re-initted in connected callback

azakus commented on packages/slider/src/mwc-slider-base.ts:
```diff
@@ -147,15 +148,15 @@ export class SliderBase extends FormElement {
       registerInteractionHandler:
           <K extends EventType>(type: K, handler: SpecificEventListener<K>) => {
-            const init = type === 'touchstart' ? {passive: true} : {};
+            const init = type === 'touchstart' ? applyPassive() : {};
```
The fallback must be `undefined` to not force `capture` behavior on IE

azakus commented on packages/slider/src/mwc-slider-base.ts:
```diff
@@ -147,15 +148,15 @@ export class SliderBase extends FormElement {
       registerThumbContainerInteractionHandler:
           <K extends EventType>(type: K, handler: SpecificEventListener<K>) => {
-            const init = type === 'touchstart' ? {passive: true} : {};
+            const init = type === 'touchstart' ? applyPassive() : {};
```
fallback to `undefined`
